katas/content/qec_shor/index.md

```diff
@@ -0,0 +1,371 @@
+<td>$1$</td>
+<td>$1$</td>
+</tr>
+<tr>
+<td>$Z_2$ (error on qubit $2$)</td>
+<td>$\alpha \ket{++-} + \beta \ket{--+}$</td>
+<td>$0$</td>
+<td>$1$</td>
+</tr>
+</table>
+
+You can see that these two parity measurements give us different pairs of results depending on whether the $Z$ error happened and on which qubit. This means that we can use them to detect the error, and then correct it by applying a $Z$ gate to the qubit that was affected by it.
+
+However, if an $X$ error happens on any one of these qubits, we won't be able to detect it: it will convert the state $\alpha \ket{+++} + \beta \ket{---}$ to the state $\alpha \ket{+++} - \beta \ket{---}$ which is a valid code word in this code - it's an encoding of the quantum state $\alpha \ket{0} - \beta \ket{1}$. We'll need to come up with a different way to detect both $X$ and $Z$ errors in the same encoding.
+
+@[exercise]({
+    "id": "qec_shor__phaseflip_encode",
+    "title": "Phase Flip Code: Encode Codewords",
+    "path": "./phaseflip_encode/",
+    "qsDependencies": [
+        "../KatasLibrary.qs"
+    ]
+})
+
+@[exercise]({
+    "id": "qec_shor__phaseflip_detect",
+    "title": "Phase Flip Code: Detect Z Error",
+    "path": "./phaseflip_detect/",
+    "qsDependencies": [
+        "../KatasLibrary.qs",
+        "./Common.qs"
+    ]
+})
+
+
+@[section]({
+    "id": "qec_shor__shor_code",
+    "title": "Shor Code"
+})
+
+Can we combine the lessons learned from the bit flip and phase flip error correction codes to be able to detect and correct both $X$ and $Z$ errors? In that case, we'd also be able to handle $Y$ errors as a combination of $X$ and $Z$ errors happening at the same time, and, as a result, we'll be able to detect and correct an arbitrary single-qubit error.
+
+Shor code, published in 1995, is the code that combines the approaches of the bit flip and phase flip codes to do just that.
+It uses the following 9-qubit encoding for logical states:
+
+$$\ket{0} \rightarrow \ket{0_L} = \frac1{2\sqrt2} (\ket{000} + \ket{111}) \otimes (\ket{000} + \ket{111}) \otimes (\ket{000} + \ket{111})$$
+$$\ket{1} \rightarrow \ket{1_L} = \frac1{2\sqrt2} (\ket{000} - \ket{111}) \otimes (\ket{000} - \ket{111}) \otimes (\ket{000} - \ket{111})$$
+$$\alpha \ket{0} + \beta \ket{1} \rightarrow \alpha \ket{0_L} + \beta \ket{1_L}$$
+
+How can we detect and correct errors using this encoding?
+
+### Detect and Correct X Errors
+
+$X$ errors happening on any qubit manifest very similarly to the way they do in the bit flip code. 
+Let's consider the first triplet of qubits and an error that happens on any of the first three qubits.
+Same as in the bit flip code, measuring the parity of pairs of qubits always returns $0$ if there is no error (since all bits in each basis state of the code words are the same), so getting parity measurement return $1$ on one or two pairs indicates an error, and the measurements which return $1$ allow us to track down the qubit on which it happened.
```

Comment:
```suggestion
Same as in the bit flip code, measuring the parity of pairs of qubits always returns $0$ if there is no error (since all bits in each basis state of the code words are the same), so a parity measurement returning $1$ on one or two pairs indicates an error, and the measurements which returned $1$ allow us to track down the qubit on which it happened.
```